### PR TITLE
Migrate to Java 17

### PR DESCRIPTION
**Pull request by gabilang**
## Purpose
$title
## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests
- [ ] Updated the spec
- [ ] Checked native-image compatibility

### CODE REVIEW

File: .github/workflows/build-timestamped-master.yml [hunk 1]
```diff
@@ -16,11 +16,11 @@ jobs:
     if: github.repository_owner == 'ballerina-platform'
     steps:
       - uses: actions/checkout@v2
-      - name: Set up JDK 11
+      - name: Set up JDK 17
         uses: actions/setup-java@v2
```
Comment (ThisaruGuruge) [inline .github/workflows/build-timestamped-master.yml:20]: I think we should use setup-java@3 for this as well.

File: .github/workflows/build-timestamped-master.yml [hunk 2]
```diff
@@ -16,11 +16,11 @@ jobs:
     if: github.repository_owner == 'ballerina-platform'
     steps:
       - uses: actions/checkout@v2
```
Comment (ThisaruGuruge) [inline .github/workflows/build-timestamped-master.yml:18]: Should migrate to checkout@v3

Comment (warunalakshitha) [inline .github/workflows/build-timestamped-master.yml:18]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/build-timestamped-master.yml (hunk 1 above)
Comment (warunalakshitha) [inline .github/workflows/build-timestamped-master.yml:20]:
```suggestion
        uses: actions/setup-java@v3
```

File: .github/workflows/publish-release.yml [hunk 3]
```diff
@@ -12,11 +12,11 @@ jobs:
 
     steps:
       - uses: actions/checkout@v2
```
Comment (warunalakshitha) [inline .github/workflows/publish-release.yml:14]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/pull-request.yml [hunk 4]
```diff
@@ -12,11 +12,11 @@ jobs:
       cancel-in-progress: true
     steps:
       - uses: actions/checkout@v2
```
Comment (warunalakshitha) [inline .github/workflows/pull-request.yml:14]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/pull-request.yml [hunk 5]
```diff
@@ -37,11 +37,11 @@ jobs:
...
```
Comment (warunalakshitha) [inline .github/workflows/pull-request.yml:39]:
```suggestion
      - uses: actions/checkout@v3
```

File: .github/workflows/publish-release.yml [hunk 6]
```diff
@@ -12,11 +12,11 @@ jobs:
...
-      - name: Set up JDK 11
+      - name: Set up JDK 17
         uses: actions/setup-java@v2
```
Comment (warunalakshitha) [inline .github/workflows/publish-release.yml:16]:
```suggestion
        uses: actions/setup-java@v3
```

File: .github/workflows/pull-request.yml [hunk 7]
```diff
@@ -12,11 +12,11 @@ jobs:
...
-      - name: Set up JDK 11
+      - name: Set up JDK 17
         uses: actions/setup-java@v2
```
Comment (warunalakshitha) [inline .github/workflows/pull-request.yml:16]:
```suggestion
        uses: actions/setup-java@v3
```

File: .github/workflows/pull-request.yml [hunk 8]
```diff
@@ -37,11 +37,11 @@ jobs:
...
```
Comment (warunalakshitha) [inline .github/workflows/pull-request.yml:41]:
```suggestion
        uses: actions/setup-java@v3
```

**Comment by warunalakshitha** [comment]
IMO we need to change all workflow actions to v3. @gabilang Shall we send separate PRs as well.

**Review by warunalakshitha — APPROVED** [review]

**Comment by gabilang** [comment]
> IMO we need to change all workflow actions to v3. @gabilang Shall we send separate PRs as well.

Ack

**Pull request merged**